benchmark/run_bench.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+
+rs=(1)
+cs=(1)
+w=10
+d=10
+reqs=(1)
+resps=(1)
+rpc_types=(unary)
+
+# idx[0] = idx value for rs
+# idx[1] = idx value for cs
+# idx[2] = idx value for reqs
+# idx[3] = idx value for resps
+# idx[4] = idx value for rpc_types
+idx=(0 0 0 0 0)
+idx_max=(1 1 1 1 1)
+
+inc()
+{
+  for i in $(seq $((${#idx[@]}-1)) -1 0)
+  do
+    idx[${i}]=$((${idx[${i}]}+1))
+    if [ ${idx[${i}]} == ${idx_max[${i}]} ]; then
+      idx[${i}]=0
+    else
+      break
+    fi
+  done
+  return=1
+  # Check to see if we have looped back to the beginning.
+  for v in ${idx[@]}; do
+    if [ ${v} == 0 ]; then
+      return=$((${return}*1))
+    else
+      return=$((${return}*0))
+    fi
+  done
+  if [ ${return} == 1 ]; then
+  rm -Rf ${out_dir}
+    exit 0
```

Comment:
`exit` terminates the script; did you want `return`?

Reply:
`inc` is called from an infinite loop. `exit 0` makes sure that we stop the script when we are done with all combinations.